akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -198,66 +199,71 @@ private[akka] class Shard(
 
   final class Entities {
 
-    private var entities: Map[EntityId, EntityState] = Map.empty
+    private val entities: java.util.Map[EntityId, EntityState] = new util.HashMap[EntityId, EntityState]()
     // needed to look up entity by reg when a Passivating is recieved
     private var byRef: Map[ActorRef, EntityId] = Map.empty
 
     def alreadyRemembered(set: Set[EntityId]): Entities = {
-      entities = set.toList.map((_, RememberedButNotCreated)).toMap
+      set.foreach { id =>
+        entities.put(id, RememberedButNotCreated)
+      }
       this
```

Comment:
One thing, since it is a mutable class, perhaps we should skip the returning of this from mutating methods (not sure we are using them anyways) since that makes it look like it could be immutable?

Reply:
👍